### PR TITLE
Decomp funcs in code_182B30.c, add some structs to common_structs.h

### PR DESCRIPTION
**Pull request by Rainchus**

### CODE REVIEW

File: include/common_structs.h [hunk 1]
```diff
@@ -1767,4 +1777,22 @@ typedef struct Temp8010EBB0 {
     /* 0x35C */ char unk_35C[0x4];
 } Temp8010EBB0; // size = 0x360
 
+typedef struct Temp8025D160 {
+    /* 0x00 */ s32 unk_00;
+    /* 0x04 */ char unk_04[8];
+    /* 0x0C */ struct Temp8025D160_2* unk_8025D160_2;
```
Comment (ethteck) [inline include/common_structs.h:1783]: could you name this field unk_0C

File: src/code_182B30.c [hunk 2]
```diff
@@ -42,17 +42,53 @@ void func_80254610(Actor* actor) {
...
+void reset_partner_blur(void) {
+    reset_actor_blur(gBattleStatus.partnerActor);
+}
 
 void func_802546B0(void) {
     func_80254610(gBattleStatus.partnerActor);
 }
 
+#ifdef NON_MATCHING
+
```
Comment (ethteck) [inline src/code_182B30.c:62]: you can remove the empty line here

**Review by ethteck — CHANGES_REQUESTED** [review]
2 minor things. other than that, looks good!

**Review by nanaian — APPROVED** [review]

File: include/common_structs.h [hunk 3]
```diff
@@ -1225,11 +1236,12 @@ typedef struct DecorationTable {
     /* 0x89C */ u8 rotZ[16];
     /* 0x8AC */ u8 effectType; /* 0 =  blur, 14 = none? */
     /* 0x8AD */ char unk_8AD[3];
-    /* 0x8B0 */ s32* unk_8B0[2];
+    /* 0x8B0 */ struct Temp8025D160* unk_8025D160[2];
```
Comment (ethteck) [inline include/common_structs.h:1239]: Can you name this field unk_8B0

File: src/code_10400_len_d30.c [hunk 4]
```diff
@@ -1,8 +1,50 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "code_10400_len_d30", state_init_language_select);
+void state_init_language_select(void) {
+    D_800A0931 = 0;
+    D_800A0932[0] = 0;
+    disable_player_input();
+    func_80027088(2);
+    set_transition_stencil_zoom_0(0, (f32) D_800A0932[0]);
```
Comment (ethteck) [inline src/code_10400_len_d30.c:8]: This cast shouldn't be necessary

File: src/code_10400_len_d30.c [hunk 5]
```diff
@@ -1,8 +1,50 @@
...
+    cameras[0].unk_06 = 1;
+    cameras[0].nearClip = 0x10;
+    cameras[0].farClip = 0x1000;
+    cameras[0].flags |= 2;
+    gCurrentCameraID = 0;
+    cameras[1].flags |= 2;
+    cameras[2].flags |= 2;
+    cameras[3].flags |= 2;
+    cameras[0].vfov = 25.0f;
+    set_cam_viewport(0, 0xC, 0x1C, 0x128, 0xB8);
```
Comment (ethteck) [inline src/code_10400_len_d30.c:32]: Here and below with the 0x64, could you change these numbers to decimal?

File: src/code_10400_len_d30.c [hunk 6]
```diff
@@ -14,9 +56,31 @@ void state_drawUI_language_select(void) {
 void state_drawUI_file_select(void) {
 }
 
-INCLUDE_ASM(s32, "code_10400_len_d30", func_800355D4);
+void func_800355D4(void) {
+    if (D_800A0932[0] > 0) {
+        D_800A0931 = 0;
+        set_transition_stencil_zoom_0(0, D_800A0932[0]);
```
Comment (ethteck) [inline src/code_10400_len_d30.c:62]: can the `set_transition_stencil_zoom_0(0, D_800A0932[0]);` line be moved after the if/else? It seems redundant to have it be in both cases exactly the same

File: src/code_10400_len_d30.c [hunk 7]
```diff
@@ -14,9 +56,31 @@ void state_drawUI_language_select(void) {
...
+void func_80035660(void) {
+    D_800A0931 = 0;
+    D_800A0932[0] = 0;
+    D_800A0930 = 0;
+    if (func_80244BC4() == 0) {
+        set_map_transition_effect(7);
+    } else {
+        set_map_transition_effect(8);
+        OVERRIDE_FLAG_UNSET(0x0010000);
+        set_music_track(0, -1, 0, 0x3E8, 8);
```
Comment (ethteck) [inline src/code_10400_len_d30.c:80]: 1000 instead of 0x3E8

File: src/code_13870_len_6980.c [hunk 8]
```diff
@@ -14,15 +14,37 @@ void mtx_ident_mirror_y(Matrix4f* mtx) {
 
 INCLUDE_ASM(s32, "code_13870_len_6980", clear_npcs);
 
-INCLUDE_ASM(s32, "code_13870_len_6980", init_npc_list);
+void init_npc_list(void) {
+    if (gGameStatusPtr[0]->isBattle == 0) {
```
Comment (ethteck) [inline src/code_13870_len_6980.c:18]: ! instead of == 0 because this is a flag

File: src/code_13870_len_6980.c [hunk 9]
```diff
@@ -14,15 +14,37 @@ void mtx_ident_mirror_y(Matrix4f* mtx) {
...
+    } else {
+        gCurrentNpcListPtr = &gBattleNpcList;
+    }
+    D_8009A604 = 0;
+    D_800A0B94 = 1;
+}
 
-INCLUDE_ASM(s32, "code_13870_len_6980", func_8003857C);
+void func_8003857C(void) {
+    s32 temp_v0;
```
Comment (ethteck) [inline src/code_13870_len_6980.c:28]: If you swap the variable order, you can move the initializations into the same line as the declarations I think.

File: src/code_13870_len_6980.c [hunk 10]
```diff
@@ -14,15 +14,37 @@ void mtx_ident_mirror_y(Matrix4f* mtx) {
...
+    s32 phi_v1;
+
+    phi_v1 = 0;
+    temp_v0 = phi_v1 < 0x40;
+    while (temp_v0 ) {
+        phi_v1 = phi_v1 + 1;
```
Comment (ethteck) [inline src/code_13870_len_6980.c:34]: You should be able to ++ here

File: src/code_13870_len_6980.c [hunk 11]
```diff
@@ -67,13 +89,67 @@ INCLUDE_ASM(Npc*, "code_13870_len_6980", get_npc_unsafe, NpcId npcId);
...
+        npc->flags = npc->flags | 0x10010;
+    }
+}
 
-INCLUDE_ASM(s32, "code_13870_len_6980", set_npc_sprite);
+void disable_npc_shadow(Npc* npc) {
+    Shadow* shadow;
 
-INCLUDE_ASM(s32, "code_13870_len_6980", enable_npc_blur);
+    if ((npc->flags & 0x10)) {
```
Comment (ethteck) [inline src/code_13870_len_6980.c:105]: You can remove one layer of these parentheses

File: src/code_13870_len_6980.c [hunk 12]
```diff
@@ -67,13 +89,67 @@ INCLUDE_ASM(Npc*, "code_13870_len_6980", get_npc_unsafe, NpcId npcId);
...
+        npc->flags &= ~0x10000;
+    }
+}
+
+func_802DE2AC(s32 arg0, s32 arg1, f32 arg2);
+
+void set_npc_sprite(Npc *npc, s32 arg1, s32 arg2) {
+    s32 flagsTemp;
+    
+
```
Comment (ethteck) [inline src/code_13870_len_6980.c:118]: We don't need two empty lines here

File: src/code_13870_len_6980.c [hunk 13]
```diff
@@ -67,13 +89,67 @@ INCLUDE_ASM(Npc*, "code_13870_len_6980", get_npc_unsafe, NpcId npcId);
...
+void enable_npc_blur(Npc *npc) {
+    NpcBlurData* blurData;
+    s32 i;
+
+    if (!(npc->flags & 0x100000)) {
+        npc->flags |= 0x100000;
+        blurData = heap_malloc(sizeof(NpcBlurData));
+        npc->blurData = blurData;
+        ASSERT(!(blurData == NULL));
```
Comment (ethteck) [inline src/code_13870_len_6980.c:142]: is `ASSERT(blurData != NULL);` possible?

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/code_10400_len_d30.c (hunk 6 above)
Comment (Rainchus) [reply src/code_10400_len_d30.c:62]: sticking it after the if/else changes the asm quite a bit

**Pull request merged**